src/query/service/src/pipelines/processors/transforms/transform_dictionary.rs

```diff
@@ -0,0 +1,129 @@
+                    DictionarySource::Redis(redis_source) => {
+                        let mut builder = Redis::default().endpoint(&redis_source.connection_url);
+                        if let Some(ref username) = redis_source.username {
+                            builder = builder.username(username);
+                        }
+                        if let Some(ref password) = redis_source.password {
+                            builder = builder.password(password);
+                        }
+                        if let Some(db_index) = redis_source.db_index {
+                            builder = builder.db(db_index);
+                        }
+                        let op = build_operator(builder)?;
+                        operators.insert(i, Arc::new(op));
+                    }
+                    DictionarySource::Mysql(_) => {
+                        return Err(ErrorCode::Unimplemented("Mysql source is unsupported"));
+                    }
+                }
+            }
+        }
+        Ok(operators)
+    }
+
+    // transform add dict get column.
+    pub(crate) async fn transform_dict_get(
+        &self,
+        i: usize,
+        data_block: &mut DataBlock,
+        dict_arg: &DictGetFunctionArgument,
+        arg_indices: &[IndexType],
+        data_type: &DataType,
+    ) -> Result<()> {
+        let op = self.operators.get(&i).unwrap().clone();
+
+        // only support one key field.
+        let arg_index = arg_indices[0];
+        let entry = data_block.get_by_offset(arg_index);
+        let value = match &entry.value {
+            Value::Scalar(scalar) => {
+                if let Scalar::String(key) = scalar {
+                    let buffer = op.read(key).await;
+                    match buffer {
+                        Ok(res) => {
+                            let value =
+                                unsafe { String::from_utf8_unchecked(res.current().to_vec()) };
+                            Value::Scalar(Scalar::String(value))
+                        }
+                        Err(_) => Value::Scalar(dict_arg.default_value.clone()),
+                    }
+                } else {
+                    Value::Scalar(dict_arg.default_value.clone())
+                }
+            }
+            Value::Column(column) => {
+                let mut builder = ColumnBuilder::with_capacity(data_type, column.len());
+                for scalar in column.iter() {
+                    if let ScalarRef::String(key) = scalar {
+                        let buffer = op.read(key).await;
+                        match buffer {
+                            Ok(res) => {
+                                let value =
+                                    unsafe { String::from_utf8_unchecked(res.current().to_vec()) };
+                                builder.push(ScalarRef::String(value.as_str()));
+                            }
+                            Err(_) => {
```

Comment:
the error must be handled.

If key is in redis but there is something wrong with redis connection, we should throw the error rather than give a default value.